Node v22 MSVC c++20 compilation failure

When building on Windows with Node v22.11.0 there is a compilation failure (this seems to occur both with VS2019 and VS2022):
```
cld\deps\cld\internal\scoreonescriptspan.h(117,16): error C7626: unnamed class used in typedef name cannot declare members other than non-static data members, member enumerations, or member classes [%ROOTPATH%\node_modules\cld\build\deps\cld\cld-c.vcxproj]
cld\deps\cld\internal\scoreonescriptspan.h(132,16): error C7626: unnamed class used in typedef name cannot declare members other than non-static data members, member enumerations, or member classes [%ROOTPATH%\node_modules\cld\build\deps\cld\cld-c.vcxproj]
cld\deps\cld\internal\scoreonescriptspan.h(186,16): error C7626: unnamed class used in typedef name cannot declare members other than non-static data members, member enumerations, or member classes [%ROOTPATH%\node_modules\cld\build\deps\cld\cld-c.vcxproj]
```

It seems to be caused by changes in Node 22 with the switch to `-std:c++20` from `-std:c++17` (https://github.com/nodejs/node/commit/b338202fab6a7d15f18918921639d16452ee964f).

Additional info:
```
node -v v22.11.0
node-gyp -v v10.2.0
```

--- a/deps/cld/internal/scoreonescriptspan.h
+++ b/deps/cld/internal/scoreonescriptspan.h
@@ -114,7 +114,7 @@ typedef struct {
 } ScoringTables;
 
 // Context for boosting several languages
-typedef struct {
+typedef struct LangBoosts_ {
    int32 n;
    uint32 langprob[kMaxBoosts];
    int wrap(int32 n) {return n & (kMaxBoosts - 1);}
@@ -129,7 +129,7 @@ typedef struct {
 
 // ScoringContext carries state across scriptspans
 // ScoringContext also has read-only scoring tables mapping grams to qprobs
-typedef struct {
+typedef struct ScoringContext_ {
   FILE* debug_file;                   // Non-NULL if debug output wanted
   bool flags_cld2_score_as_quads;
   bool flags_cld2_html;
@@ -183,7 +183,7 @@ typedef struct {
 } LangprobHit;
 
 // Holds arrays of scoring-table lookup hits for (part of) a scriptspan
-typedef struct {
+typedef struct ScoringHitBuffer_ {
   ULScript ulscript;        // langprobs below are with respect to this script
   int maxscoringhits;       // determines size of arrays below
   int next_base;            // First unused entry in each array

